src/test/lrsql/admin/protocol_test.clj

```diff
@@ -155,6 +183,49 @@
         (is (-> (adp/-ensure-account-oidc lrs username bad-issuer)
                 :result
                 (= :lrsql.admin/oidc-issuer-mismatch-error)))))
+    (testing "actor-delete"
+      (let [stmts [stmt-0 stmt-1 stmt-2 stmt-3]
+            ifis (->> (conj stmts (stmt-3 "object"))
+                      (map #(ua/actor->ifi (% "actor"))))
+            get-stmt-#s (fn []
+                          (reduce (fn [m actor-ifi]
+                                    (assoc m actor-ifi
+                                           (count (:statements (:statement-result (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []))))))
+                                  {} ifis))]
+        (lrsp/-store-statements lrs auth-ident stmts [])
+
+        (assert (every? #(> % 0) (vals (get-stmt-#s))))
+        (doseq [ifi ifis]
+          (adp/-delete-actor lrs {:actor-ifi ifi}))
+        (is (every? #(= % 0) (vals (get-stmt-#s))))))
+    (let [stmt->ifi #(ua/actor->ifi (% "actor"))
+          count-of-actor (fn [actor-ifi] (-> (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []) :statement-result :statements count))
+          child-ifi (stmt->ifi (stmt-3 "object"))
+          parent-ifi (stmt->ifi stmt-3)]
+      (testing "delete-actor correctly deletes statements that are parent to actor (sub)statements"
+        (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+        (assert (= 1 (count-of-actor parent-ifi)))
+        (adp/-delete-actor lrs {:actor-ifi child-ifi})
+        (is (= 0 (count-of-actor parent-ifi)));
+        (adp/-delete-actor lrs {:actor-ifi parent-ifi}))
+      (testing "delete-actor correctly deletes substatements that are child to actor statements"
+        (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+        (assert (= 1 (count-of-actor child-ifi)))
+        (adp/-delete-actor lrs {:actor-ifi parent-ifi})
+        (is (= 0 (count-of-actor child-ifi)))
+        (adp/-delete-actor lrs {:actor-ifi child-ifi}))
+      (testing "for StatementRefs, delete-actor deletes statement->actor relationships but leaves statements by another actor untouched"
+        (let [[ifi-0 ifi-2] (mapv stmt->ifi [stmt-0 stmt-2])]
+          (lrsp/-store-statements lrs auth-ident [stmt-0] [])
+          (lrsp/-store-statements lrs auth-ident [stmt-2] [])      
```

Comment:
this could be a single call, might make this a little easier to read.

Reply:
:+1: